ext/lmdb-safe/lmdb-safe.cc

```diff
@@ -16,6 +21,56 @@ static string MDBError(int rc)
   return mdb_strerror(rc);
 }
 
+#ifndef DNSDIST
+
+namespace LMDBLS {
+  // this also returns a pointer to the string's data. Do not hold on to it too long!
+  LSheader* LSassertFixedHeaderSize(std::string_view val) {
+    // cerr<<"val.size()="<<val.size()<<endl;
+    if (val.size() < LS_MIN_HEADER_SIZE) {
+      throw std::runtime_error("LSheader too short");
+    }
+
+    return (LSheader*)val.data();
+  }
+
+  size_t LScheckHeaderAndGetSize(std::string_view val, size_t datasize) {
+    LSheader* lsh = LSassertFixedHeaderSize(val);
+
+    if (lsh->d_version != 0) {
+      throw std::runtime_error("LSheader has wrong version (not zero)");
+    }
+
+    size_t headersize = LS_MIN_HEADER_SIZE;
+
+    headersize += ntohs(lsh->d_numextra) * LS_BLOCK_SIZE;
```

Comment:
I suppose we trust the LMDB file? Otherwise this need an overflow guard

Reply:
we do (the existing code will also crash if there is garbage in the file), but over time I would indeed like to be more robust against these cases

Reply:
but, in this case there's an `if` right below, is it missing a case?